charts/common/templates/_names.tpl

```diff
@@ -0,0 +1,99 @@
+{{/* vim: set filetype=mustache: */}}
+
+{{/*
+Get postgres host name in the following order of precendence: Chart specific postgresql > Chart external Postgresql > Global postgresql, Arguments:
+1) postgresql object
+2) external postgresql object
+3) Release object
+4) Chart object
+*/}}
+{{- define "common.postgres-host" -}}
+{{- $postgresql := index . 0 -}}
+{{- $externalPostgresql := index . 1 -}}
+{{- $release := index . 2 -}}
+{{- $chart := index . 3 -}}
+{{- if $postgresql.enabled -}}
+    {{- printf "%s-%s-postgresql.%s.svc.cluster.local" $release.Name $chart.Name $release.Namespace -}}
+{{- else if $externalPostgresql.enabled -}}
+    {{- $externalPostgresql.address -}}
+{{- else -}}
+    {{- printf "%s-postgresql.%s.svc.cluster.local" $release.Name $release.Namespace -}}
```

Comment:
Just for my understanding, in what scenarios will the first 2 branches be skipped and the else branch executes?
```
{{- else -}}
 {{- printf "%s-postgresql.%s.svc.cluster.local" $release.Name $release.Namespace -}}
 ```

Reply:
for the case of the umbrella chart. We create a Postgres in the caraml chart to be used by every component.